css/css-anchor-position/support/test-common.js

```diff
@@ -25,3 +25,23 @@ function assert_fallback_position(anchored, anchor, direction) {
 async function waitUntilNextAnimationFrame() {
   return new Promise(resolve => requestAnimationFrame(resolve));
 }
+
+// This function is a thin wrapper around `checkLayout` (from
+// resources/check-layout-th.js) and simply reads the `CHECK_LAYOUT_DELAY`
+// variable to add a delay. This global variable is not intended to be set by
+// browsers; instead, polyfills can set it to give themselves time to apply
+// changes before proceeding with assertions about the layout. Tests that call
+// this function at some point and do some other work after should `await` it to
+// avoid race conditions.
+window.checkLayoutForAnchorPos = async function(selectorList, callDone = true) {
+  if (window.CHECK_LAYOUT_DELAY) {
+    if (!window.INJECTED_SCRIPT) {
+      throw new Error('CHECK_LAYOUT_DELAY is only allowed when serving WPT with --injected-script.')
+    }
+    console.log(`Waiting before checking layout...`);
+    await waitUntilNextAnimationFrame();
+    await waitUntilNextAnimationFrame();
+    await waitUntilNextAnimationFrame();
```

Comment:
@mfreed7 I tried waiting for 2 frames, but I was still hitting race conditions. Three seem to do the trick